go/pkg/pwapi/api_challenge-subscription-validate.go

```diff
@@ -53,7 +53,16 @@ func (svc *service) ChallengeSubscriptionValidate(ctx context.Context, in *Chall
 	// compare input and instances' passphrases
 	var amountExpected int
 	{
-		configData, err := instances[0].ParseInstanceConfig()
+		var availableInstance *pwdb.ChallengeInstance
+		for _, instance := range instances {
+			if instance.Status == pwdb.ChallengeInstance_Available {
+				availableInstance = instance
+			}
+		}
+		if availableInstance == nil {
+			return nil, errcode.ErrNoAvailableChallengeInstance
+		}
```

Comment:
this part can works, but if this is only for this part, I prefer to use a `found := false` check